data-protocols/ids/ids-core/src/main/java/org/eclipse/dataspaceconnector/ids/core/message/DataRequestMessageSender.java

```diff
@@ -138,12 +155,32 @@ public DataRequestMessageSender(String connectorId,
         return future;
     }
 
+    private RejectionMessage extractRejectionMessage(Response r) {
+        try {
+            return mapper.readValue(r.body().string(), RejectionMessage.class);
+        } catch (IOException ex) {
+            monitor.severe("Could not read body of response", ex);
+            return null;
+        }
+    }
+
+    private Object extractArtifactResponse(Response httpResponse) {
+        try {
+            var body = httpResponse.body().string();
+            var artifactResponse = mapper.readValue(body, ArtifactResponseMessage.class);
+            return artifactResponse.getProperties().get("dataspaceconnector-data-object");
+        } catch (IOException e) {
+            monitor.severe("Could not read body of response", e);
+            return null;
+        }
+    }
+
```

Comment:
In case of IDS Rest there is no IDS message in the response body; the information will be in the headers. Nothing that has to be fixed now, but we should keep this in mind. Can you add a TODO/fixme?

Reply:
will do.

Reply:
When we abandon IDS enirely, we'll likely have to re-implement this feature in IDS multipart anyway. But that'll be done in a subsequent PR.